actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -122,6 +122,7 @@ def find_routes req
 
         def match_head_routes(routes, req)
           head_routes = match_routes(routes, req)
+          head_routes.delete_if { |route| route.verb == // }
```

Comment:
@tgxworld yep, I think that should be okay - @tsun1215 can you change the PR please.